cmd/clusterctl/examples/ibmcloud/generate-yaml.sh

```diff
@@ -112,8 +112,8 @@ cat "$WORKER_USER_DATA" > $USERDATA/$PROVIDER_OS/worker-user-data.sh
 
 # Set up the output dir if it does not yet exist
 mkdir -p $PWD/$OUTPUT
-cp -n $PWD/cluster.yaml.template $PWD/$OUTPUT/cluster.yaml
-cp -n $PWD/machines.yaml.template $PWD/$OUTPUT/machines.yaml
+cp -n $PWD/cluster.yaml.template $PWD/$OUTPUT/cluster.yaml || true
+cp -n $PWD/machines.yaml.template $PWD/$OUTPUT/machines.yaml || true
```

Comment:
should we always overwrite the file?

Reply:
I think we should not overwrite, as the overwrite will use template to overwrite which will clear some useful data.

Reply:
the `cluster.yaml` and `machines.yaml` only contain some cluster info and machine info which should matainered by cluster deployer self, so I think we can only overwrite the `provider-components.yaml` when use overwrite flag.

cc @gyliu513 comments?

Reply:
`provider-components.yaml` will always overwrite, this is the current logic already. I think the current logic is ok. @hchenxa @jichenjc